### DISCUSSION TITLE
Realtime filter with OR / multiple columns.

### DISCUSSION THREAD

**Discussion by Danieltech99** [supabase/supabase#1791] [Questions] [answered]
I am currently working on a real time chat. Messages have a `to_id` and `from_id`.

For select I was able to use the following simplified query (as per PostgREST): 
```typscript
query.or(`from_id.eq.${id},to_id.eq.${id}`);
```
Which translates to `or(from_id.eq.${id},to_id.eq.${id})`.

But how can I use an `or` statement to filter for realtime subscriptions? Ideally, I would like to be able to use `or` and `and` so that I can match both pairs: `or( and(from_id.eq.${id1},to_id.eq.${id2}) , and(from_id.eq.${id2},to_id.eq.${id1}) )`. I see that there is a way to filter on a single column but I can't seem to see how to filter on multiple columns or use an `or` clause. This works `friends_messages:to_id=eq.${id}` but is incomplete. I tried `friends_messages:or=(from_id.eq.${id},to_id.eq.${id})` but it did not work. This is for the chat page, so I don't want all messages.

**Comment by steve-chavez** [comment, marked as answer]
Hey @Danieltech99,

Yes, realtime only supports the `eq` filter for now(see [docs](https://github.com/supabase/realtime-js/#subscribing-to-events)). I suggest you open an issue requesting the feature on [realtime-js](https://github.com/supabase/realtime-js).
